File: backend/app/schemas/requirement.py
"""需求文档模块的 Pydantic 数据模型。"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RequirementOut(BaseModel):
    """需求文档列表项。"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    file_name: str
    file_type: str
    file_size: int
    parse_status: str
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class RequirementDetailOut(RequirementOut):
    """需求文档详情（含提取的文本内容）。"""

    content: str | None


class RequirementListOut(BaseModel):
    """需求文档分页列表响应。"""

    total: int
    page: int
    page_size: int
    items: list[RequirementOut]


class ParseResultOut(BaseModel):
    """AI 解析结果响应。"""

    requirement_id: int
    parse_status: str
    error_message: str | None
    result: dict | None

File: frontend/src/types/requirement.ts
// 需求文档相关类型定义
export interface Requirement {
  id: number
  project_id: number
  file_name: string
  file_type: string
  file_size: number
  parse_status: string
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface RequirementDetail extends Requirement {
  content: string | null
}

export interface RequirementListResult {
  total: number
  page: number
  page_size: number
  items: Requirement[]
}

// ---------- AI 解析结果 ----------
export interface RequirementModule {
  name: string
  description: string
  functions: string[]
}

export interface BusinessFlow {
  name: string
  steps: string[]
}

export interface RiskPoint {
  type: string
  description: string
  level: string
}

export interface ParseResultData {
  summary: string
  modules: RequirementModule[]
  roles: string[]
  business_flows: BusinessFlow[]
  risks: RiskPoint[]
}

export interface ParseResultResponse {
  requirement_id: number
  parse_status: string
  error_message: string | null
  result: ParseResultData | null
}
